core/lib/spree/core/controller_helpers/pricing.rb

```diff
@@ -7,12 +7,46 @@ module Pricing
         extend ActiveSupport::Concern
 
         included do
-          helper_method :current_pricing_options
+          # Key where the current currency should be stored in a requests' session
+          CURRENCY_SESSION_KEY = 'currency'
+
+          helper_method :current_pricing_options,
+                        :current_currency,
+                        :switch_currency
         end
 
+        # @see Variant::PricingOptions#from_context
         def current_pricing_options
           Spree::Config.pricing_options_class.from_context(self)
         end
+
+        # Current currency from {#current_pricing_options}
+        #
+        # @return [String]
+        # @see #current_pricing_options
+        # @example
+        #   'USD'
+        def current_currency
+          current_pricing_options.currency
+        end
+
+        # Currency stored in the session
+        #
+        # @return [String, nil]
+        # @see CURRENCY_SESSION_KEY
+        # @example
+        #   'USD'
+        def currency_in_session
+          session[CURRENCY_SESSION_KEY]
+        end
+
+        # Stores given currency in the session
+        #
+        # @param currency [String]
+        # @see #currency_in_session
+        def switch_currency(currency)
+          session[CURRENCY_SESSION_KEY] = currency
+        end
```

Comment:
What about creating a new object to store the currency information that will be used here? Something like:

```ruby
class Spree::Currency::SessionStore
  CURRENCY_SESSION_KEY = 'currency'

  def initialize(context)
    @session = context.session
  end

  def currency
    @session[CURRENCY_SESSION_KEY]
  end

  def currency=
    @session[CURRENCY_SESSION_KEY] = currency
  end
end
```

Setting the currency store in a preference like

```ruby
Spree::Config.currency_store = Spree::Currency::SessionStore
```

And finally, use it in the pricing helper as:

```ruby
        def currency_in_session
          currency_store.currency
        end

        def switch_currency(currency)
          currency_store.currency = currency
        end

        private

        def currency_store
          @currency_store ||= Spree::Config.currency_store.new(self)
        end
```

The benefit of this approach is that it allows people to change the way currency is stored. They would just need to create their own "currency store" class that responds to `currency` and `currency=` and change the preference.

Reply:
Undoubtedly it would add more flexibility. But don't you think it would be overkilling? Do you think we need to provide this flexibility at this point? I'm open to implementing it, it's just that I'm considering complexity as a trade-off.

Reply:
You are probably right, I also don't want to anticipate a design before we actually need it. I'm ok with the current version, we can always change this later if needed.

Reply:
To be honest, I have been triggered by `currency_in_session` method name, which seems too tied to the implementation. Maybe it's a good thing to be explicit but in this case shouldn't we use the same approach when we set the currency, so using something like `switch_currency_in_session`?

Reply:
I had the same internal conversation about it :wink: But I decided to go with this version because `switch_currency` is in fact changing the end currency seen by the user (which is indeed the same that it's in the session), while the reader method `currency_in_session` needs to be differentiated from `current_currency`, which will take the one from the session only when it's set. WDYT?